x/registry/types/message_buy_alias.go

```diff
@@ -0,0 +1,62 @@
+package types
+
+import (
+	"regexp"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const TypeMsgBuyAlias = "buy_alias"
+
+var _ sdk.Msg = &MsgBuyAlias{}
+
+func NewMsgBuyAlias(creator string, did string, name string) *MsgBuyAlias {
+	return &MsgBuyAlias{
+		Creator: creator,
+		Did:     did,
+		Name:    name,
+	}
+}
+
+func (msg *MsgBuyAlias) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgBuyAlias) Type() string {
+	return TypeMsgBuyAlias
+}
+
+func (msg *MsgBuyAlias) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgBuyAlias) GetSignBytes() []byte {
```

Comment:
nit: Signed?

Reply:
This is implementation method for the underlying cosmos message - cant change function name